### PR TITLE
BUG: MultiIndex.unique losing ea dtype

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

the MultiIndex.unique code path is casting the values to tuples and runs through unique1d. This is really slow.

```
       before           after         ratio
     [e7414aa1]       [d17eb141]
     <48184^2>        <perf_mi_unique>
-         422±2ms       30.3±0.5ms     0.07  multiindex_object.Unique.time_unique_dups(('Int64', <NA>))
-        496±70ms         28.4±1ms     0.06  multiindex_object.Unique.time_unique_dups(('int64', 0))
-         811±4ms       32.9±0.2ms     0.04  multiindex_object.Unique.time_unique(('int64', 0))
-        812±20ms       32.8±0.9ms     0.04  multiindex_object.Unique.time_unique(('Int64', <NA>))
```

### CODE REVIEW

File: pandas/tests/indexes/multi/test_duplicates.py [hunk 1]
```diff
@@ -337,3 +338,16 @@ def test_multi_drop_duplicates_pos_args_deprecation():
...
+
+
+def test_midx_unique_ea_dtype():
+    # GH#48335
+    a = Series([1, 2, NA, NA], dtype="Int64")
+    b = np.array([1, 2, 3, 3])
+    midx = MultiIndex.from_arrays([a, b], names=["a", "b"])
+    result = midx.unique()
+
+    a = Series([1, 2, NA], dtype="Int64")
```
Comment (jbrockmendel) [inline pandas/tests/indexes/multi/test_duplicates.py:350]: nitpick, can you avoid 1-letter variable names

Comment (phofl) [reply pandas/tests/indexes/multi/test_duplicates.py]: Yep, thx. Stupid copy paste miss

**Comment by phofl** [comment]
cc @mroeschke for a second set of eyes

File: pandas/core/indexes/multi.py [hunk 2]
```diff
@@ -1728,7 +1728,7 @@ def get_level_values(self, level):
     def unique(self, level=None):
 
         if level is None:
-            return super().unique()
```
Comment (mroeschke) [inline pandas/core/indexes/multi.py:1731]: Looks like this old path would trigger a `_shallow_copy` and it doesn't look like `drop_duplicates` does. Any implications off the top of your head? (Maybe another reason why there's a perf improvement)

Comment (phofl) [reply pandas/core/indexes/multi.py:1731]: Hm not really, we are returning a view on the original data anyway if the MultiIndex is unique, so the duplicate case does not have to trigger a _shallow_copy I think.

The performance improvement has another cause:

Before this, we cast our values to a numpy array of tuples. This pr avoids that.

**Review by mroeschke — APPROVED** [review]
LGTM

**Pull request merged**